samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -31,9 +41,15 @@ def normalize(template_dict):
 
         for logical_id, resource in resources.items():
             resource_metadata = resource.get(METADATA_KEY, {})
-            asset_path = resource_metadata.get(ASSET_PATH_METADATA_KEY)
             asset_property = resource_metadata.get(ASSET_PROPERTY_METADATA_KEY)
 
+            if asset_property == IMAGE_ASSET_PROPERTY:
+                asset_metadata = ResourceMetadataNormalizer._extract_image_asset_metadata(resource_metadata)
+                ResourceMetadataNormalizer._update_resource_image_asset_metadata(resource_metadata, asset_metadata)
+                asset_path = logical_id.lower()
```

Comment:
Why is the logical_id the `asset_path`?

Reply:
This is equivalent to the name of the docker image generated by sam build and sets it to the value of Code.ImageUri.